apiserver/tools.go

```diff
@@ -49,12 +51,18 @@ func (h *toolsDownloadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 		tarball, err := h.processGet(r, st)
 		if err != nil {
 			logger.Errorf("GET(%s) failed: %v", r.URL, err)
-			sendError(w, errors.NewBadRequest(err, ""))
+			if err := sendError(w, errors.NewBadRequest(err, "")); err != nil {
```

Comment:
We seem to have much copying of sendError returning an error that we just log. Wouldn't it make more sense for sendError to just log the error itself and not return an error?